### ISSUE TITLE
Documentation: Explain why you can't use the `CREATE` statement to create a relation

### ISSUE THREAD

**Issue by MatthewAry** [surrealdb/www.surrealdb.com#337]
### Description

[I recently posted a question to Discord](https://discord.com/channels/902568124350599239/1018618253695795261/1156329012457648278)

-----
So I've run into a curious behavior that I would like to understand. It seems that
```sql
CREATE projectUser SET in = project:r3dyjmgco14t93p1ga84, out = userAccount:v2uuten5fk4se471kzag;
```
and
```sql
RELATE ONLY project:r3dyjmgco14t93p1ga84->projectUser->userAccount:zzcdk6et3q74lgh409b7;
```
do not produce the same results. Because when I do
```sql
SELECT ->projectUser as users FROM $project;
```
I only get 1 result instead of 2. If I run
```sql
SELECT * FROM projectUser WHERE in = project:r3dyjmgco14t93p1ga84;
```
I get 2 results. What is `RELATE` doing that is different from my `CREATE` statement?

----

The answer I got is that there `RELATE` is not simply syntactic sugar but is actually needed to enable graph traversal functionality. This important information is absent from the documentation as far as I can tell. I request that either the [CREATE Statement](https://surrealdb.com/docs/surrealql/statements/create) and or the [RELATE statement](https://surrealdb.com/docs/surrealql/statements/select) documentation gets updated to include this important bit of information. It would also be nice to have some sort of blog post or additional information posted somewhere that explains what RELATE is doing under the hood.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Ekwuno** [comment]
Hey @MatthewAry I think the  [`RELATE` ](https://surrealdb.com/docs/surrealql/statements/relate) Statement docs highlights this functionality in detail already. 

As this extra shows 
> The RELATE statement can be used to generate graph edges between two records in the database.
   This allows you to traverse related records efficiently without needing to pull data from multiple tables and merging that 
   data together using SQL JOINs.
   
   Is there any specific type of content you are looking for?

**Comment by MatthewAry** [comment]
Yes I saw this. However there is nuance that needs to be explained.
- Is a relation "table" a variant or a superset of a normal table?
- Why can I define a table that is used for a relation and define field with assert statements, but the the relation table can't do the same things as a table?
- See https://github.com/surrealdb/surrealdb/issues/1424
- See https://discord.com/channels/902568124350599239/1157660156180574238 <- this is a good one. Why can't an edge "table" or record have another edge table/record connected to it?

In other words, I believe there should be documentation that gives more information to clearly explain the functional differences between a relation table and a normal table, because I only found out with experimentation that you can't use the CREATE statement to make a record on a table that has `id`, `in` and `out` fields and be able to use the SurrealQL's graph traversal notation, even though from an outside perspective, it looks like it's making a normal record.
